src/storageDataGroup.js

```diff
@@ -182,6 +184,93 @@ function recordsToKeys(records) {
     return keynames;
 }
 
+function waitForCompletion(path, remainingRetries) {
+    const checkOptions = {
+        host: 'localhost',
+        port: 8100,
+        path,
+        method: 'GET',
+        why: 'checking for task completion',
+        headers: {
+            Authorization: `Basic ${Buffer.from('admin:').toString('base64')}`,
+            'Content-Type': 'application/json'
+        }
+    };
+
+    return Promise.resolve()
+        .then(() => iControlRestRequest(checkOptions))
+        .then((response) => {
+            if (response.body._taskState === 'VALIDATING') {
+                if (remainingRetries > 0) {
+                    return promiseUtil.delay(500)
+                        .then(() => waitForCompletion(path, remainingRetries - 1));
+                }
+                throw new Error('Configuration save taking longer than expected');
+            }
+
+            if (response.body._taskState === 'FAILED') {
+                throw new Error(`Configuration save failed during execution: ${JSON.stringify(response.body)}`);
+            }
+
+            return Promise.resolve();
+        })
+        .catch((error) => {
+            function isAllowedError() {
+                if (error.message.indexOf('TimeoutException') > -1) {
+                    return true;
+                }
+
+                if (error.message.indexOf('response=400') > -1) {
+                    return true;
+                }
+
+                return false;
+            }
+
+            if (remainingRetries > 0 && isAllowedError()) {
+                return promiseUtil.delay(500)
+                    .then(() => waitForCompletion(path, remainingRetries - 1));
+            }
+            throw error;
+        });
+}
+
+function iControlRestRequest(opts, payload) {
```

Comment:
Can we use [atg-shared-utilities/request](https://github.com/f5devcentral/f5-atg-shared-utilities/blob/main/src/request.js) here?

Reply:
Good idea! Conversion is underway